src/__test__/AbstractAppliance.test.js

```diff
@@ -0,0 +1,96 @@
+import { AssertionError } from 'assert'
+import AbstractAppliance from '../AbstractAppliance'
+import {
+	AbstractInstantiationError,
+	NotImplementedError,
+} from '../errors'
+import Payload from '../Payload'
+import {
+	FullyImplementedAppliance,
+	PartiallyImplementedAppliance,
+} from './classes'
+
+describe('AbstractAppliance #unit', () => {
+	describe('construction', () => {
+		it('should throw an error when called directly', () => {
+			expect(() => {
+				new AbstractAppliance() // eslint-disable-line no-new
+			}).toThrow(AbstractInstantiationError)
+		})
+
+		it('should throw an error when getInputTypes() is called without implementation', () => {
+			const implementedAppliance = new PartiallyImplementedAppliance()
+			expect(() => implementedAppliance.getInputTypes()).toThrow(NotImplementedError)
+		})
+
+		it('should throw an error when getOutputTypes() is called without implementation', () => {
+			const implementedAppliance = new PartiallyImplementedAppliance()
+			expect(() => implementedAppliance.getOutputTypes()).toThrow(NotImplementedError)
+		})
+
+		it('should throw an error when isValidPayload() is called without implementation', async () => {
+			const implementedAppliance = new PartiallyImplementedAppliance()
+			await expect(async () => implementedAppliance.isValidPayload())
+				.rejects.toBeInstanceOf(NotImplementedError)
+		})
+
+		it('should throw an error when invoke() is called without implementation', async () => {
+			const implementedAppliance = new PartiallyImplementedAppliance()
+			await expect(async () => implementedAppliance.invoke())
+				.rejects.toBeInstanceOf(NotImplementedError)
+		})
+
+		it('should allow construction when extended', () => {
+			expect(() => {
+				new PartiallyImplementedAppliance() // eslint-disable-line no-new
+			}).not.toThrow(Error)
+		})
```

Comment:
While I respect the thoroughness of including this test, I'm not sure when it would ever fail us without also breaking all the other tests of the `PartiallyImplementedAppliance` above? You could safely remove it, is what I'm thinking, but your call.

Reply:
I think I wrote this one first (despite its order!)